### ISSUE TITLE
Adding UNIQUE index causes comparison operator to not work

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#7820]
## Expected behavior  
The following two queries are equivalent and should always return the same results:
```
SELECT * FROM Foo WHERE name >= "0"
SELECT * FROM Foo WHERE "0" <= name
``` 

## Actual behavior  
With a `UNIQUE` index on `Foo.name`, the second query always returns no results, whereas the first query continues working.  

## Steps to reproduce  
```
CREATE CLASS Foo EXTENDS V
CREATE PROPERTY Foo.name String

CREATE VERTEX Foo SET name = "1"
CREATE VERTEX Foo SET name = "2"

SELECT * FROM Foo WHERE name >= "0"
SELECT * FROM Foo WHERE "0" <= name
# both queries return the same results

CREATE INDEX Foo.name UNIQUE

SELECT * FROM Foo WHERE name >= "0"
# returns two rows, same as before

SELECT * FROM Foo WHERE "0" <= name
# returns no rows
```

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

Thank you very much for reporting, I'll check it today

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

I just pushed a fix on 2.2.x, it will be released with 2.2.30.
Leaving open to port the fix on 3.0

Thanks

Luigi

**Comment by obi1kenobi** [comment]
Thank you for the prompt fix, looking forward to 2.2.30!

P.S.: You mentioned you wanted to keep the issue open while you check on 3.0, but I think Github closed the issue automatically when you mentioned it in a commit that got merged. Just wanted to point that out in case that wasn't what you had in mind.

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

Yes, we set `develop` branch as default branch, so when I commit something with `Resolves: <issueNumber>` on that branch GitHub closes it for me.
Last commit was on develop branch, so everything is ok ;-)

Thanks

Luigi
